File: backend/src/main/java/com/familyhome/entity/Bill.java
package com.familyhome.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/** 账单实体：对应 t_bill 表，记录一笔支出/收入/转账流水 */
@Data
@TableName("t_bill")
public class Bill {
    /** 主键 */
    @TableId(type = IdType.AUTO)
    private Long id;
    /** 所属账本 ID */
    private Long ledgerId;
    /** 账单类型：expense(支出)/income(收入)/transfer(转账) */
    private String type;
    /** 分类 ID */
    private Long categoryId;
    /** 记账人（家庭成员 ID） */
    private Long memberId;
    /** 账单金额 */
    private BigDecimal amount;
    /** 记账时间（秒级精度） */
    private LocalDateTime billDate;
    /** 备注 */
    private String remark;
    /** 创建人 ID */
    private Long createdBy;
    /** 创建时间 */
    private LocalDateTime createdAt;
    /** 更新时间 */
    private LocalDateTime updatedAt;
}

File: frontend/src/api/bill.ts
import { http } from './http'
import type { CategoryVO, TagVO } from './ledger'

export interface BillAccountItem {
  accountId: number
  direction: 'out' | 'in'
  amount: number
}

export interface BillVO {
  id: number
  ledgerId: number
  type: 'expense' | 'income' | 'transfer'
  categoryId?: number
  categoryName?: string
  categoryIcon?: string
  memberId: number
  memberName?: string
  amount: number
  billDate: string
  remark?: string
  tags: TagVO[]
  accounts: { accountId: number; accountName: string; direction: string; amount: number; pairId?: number }[]
  createdAt: string
  updatedAt: string
}

/** 账单操作留痕 */
export interface BillLogVO {
  id: number
  action: 'create' | 'update' | 'delete'
  operatorName: string
  summary: string
  accountIcons?: string[]
  createdAt: string
}

export interface PageResult<T> {
  list: T[]
  total: number
  page: number
  size: number
}

export function listBills(
  ledgerId: number,
  params: Record<string, string | number | undefined>
): Promise<PageResult<BillVO>> {
  return http.get(`/ledgers/${ledgerId}/bills`, { params })
}

export function createBill(
  ledgerId: number,
  data: {
    type: 'expense' | 'income' | 'transfer'
    categoryId?: number
    memberId?: number
    amount: number
    billDate?: string
    remark?: string
    tagIds?: number[]
    items: BillAccountItem[]
  }
): Promise<BillVO> {
  return http.post(`/ledgers/${ledgerId}/bills`, data)
}

export function getBill(id: number): Promise<BillVO> {
  return http.get(`/bills/${id}`)
}

export function updateBill(id: number, data: any): Promise<BillVO> {
  return http.put(`/bills/${id}`, data)
}

export function deleteBill(id: number): Promise<void> {
  return http.delete(`/bills/${id}`)
}

export function billLogs(id: number): Promise<BillLogVO[]> {
  return http.get(`/bills/${id}/logs`)
}

export type { CategoryVO }
